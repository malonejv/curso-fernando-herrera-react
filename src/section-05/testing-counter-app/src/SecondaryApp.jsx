import PropTypes from "prop-types";

const SecondaryApp = ({
  quote= "The only limit to our realization of tomorrow is our doubts of today.",
  author= "Franklin D. Roosevelt",
  date= new Date(1945, 4, 13)
}) => {
  return (
    <div>
      <h1 data-testid="test-title">Hello, World!</h1>
      <blockquote>
        {quote}
        <p>â€” {author}</p>
        <p>Date: {date.toLocaleDateString()}</p>
      </blockquote>
    </div>
  );
};

SecondaryApp.propTypes = {
  quote: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date)
};

export default SecondaryApp;
