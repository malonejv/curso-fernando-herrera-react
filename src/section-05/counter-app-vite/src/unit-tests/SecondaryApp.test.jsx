import { render, screen } from "@testing-library/react";
import SecondaryApp from "../SecondaryApp";

describe("Component: SecondaryApp", () => {
  const expectedTitle = "Hello, World!";
  const expectedAuthor = "â€” Franklin D. Roosevelt";
  const expectedQuote = "Some inspiring quote here";

  test("should match snapshot", () => {
    const { container } = render(<SecondaryApp quote={expectedQuote} />);
    expect(container).toMatchSnapshot();
  });

  test("should have a test Hello World", () => {
    render(<SecondaryApp />);
    expect(screen.getByRole("heading", { level: 1, name: expectedTitle })).toBeTruthy();
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(expectedTitle);
  });

  test("should display the quote sent by props", () => {
    render(<SecondaryApp quote={expectedQuote} />);
    expect(screen.getByText(expectedQuote)).toBeTruthy();
  });
});
