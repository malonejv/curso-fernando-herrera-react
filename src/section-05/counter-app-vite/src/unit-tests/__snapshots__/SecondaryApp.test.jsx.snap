// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SecondaryApp should match snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      Hello, World!
    </h1>
    <blockquote>
      Some inspiring quote here
      <p>
        â€” 
        Franklin D. Roosevelt
      </p>
      <p>
        Date: 
        13/5/1945
      </p>
    </blockquote>
  </div>
</div>
`;
